relay-event-normalization/src/normalize/span/description/mod.rs

```diff
@@ -178,34 +179,89 @@ fn scrub_redis_keys(string: &str) -> Option<String> {
-        string = &string[..pos];
-    }
-    match RESOURCE_NORMALIZER_REGEX.replace_all(string, "$pre*$post") {
-        Cow::Owned(scrubbed) => Some(scrubbed),
-        Cow::Borrowed(string) => {
-            // No IDs scrubbed, but we still want to set something.
-            // If we managed to parse the URL, we'll try to get an extension.
-            if let Ok(url) = Url::parse(string) {
-                let domain = normalize_domain(&url.host()?.to_string(), url.port())?;
-                // If there is an extension, we add it to the domain.
-                if let Some(extension) = url.path().rsplit_once('.') {
-                    return Some(format!("{domain}/*.{}", extension.1));
-                }
-                return Some(format!("{domain}/*"));
+static DUMMY_URL: Lazy<Url> = Lazy::new(|| "http://replace_me".parse().unwrap());
+
+fn scrub_resource(string: &str) -> Option<String> {
+    let (url, is_relative) = match Url::parse(string) {
+        Ok(url) => (url, false),
+        Err(url::ParseError::RelativeUrlWithoutBase) => {
+            // Try again, with base URL
+            match Url::options().base_url(Some(&DUMMY_URL)).parse(string) {
+                Ok(url) => (url, true),
+                Err(_) => return None,
             }
-            Some(string.into())
         }
+        Err(_) => {
+            return None;
+        }
+    };
+
+    let mut formatted = match url.scheme() {
+        "data" => match url.path().split_once(';') {
+            Some((ty, _data)) => format!("data:{ty}"),
+            None => "data:*/*".to_owned(),
+        },
+        "chrome-extension" => {
+            let path = scrub_resource_path(url.path());
+            format!("chrome-extension://*{path}")
+        }
+        scheme => {
+            let path = url.path();
+            let path = scrub_resource_path(path);
+            let domain = url
+                .domain()
+                .and_then(|d| normalize_domain(d, url.port()))
+                .unwrap_or("".into());
+            format!("{scheme}://{domain}{path}")
+        }
+    };
+
+    if is_relative {
+        formatted = formatted.replace("http://replace_me", "");
     }
+
+    Some(formatted)
+}
+
+fn scrub_resource_path(string: &str) -> String {
+    let mut segments = vec![];
+    let mut extension = "";
+    for segment in string.split('/') {
+        let (base, ext) = match segment.rsplit_once('.') {
+            Some(p) => p,
+            None => (segment, ""),
+        };
+        extension = ext;
+        let base = RESOURCE_NORMALIZER_REGEX.replace_all(base, "$pre*$post");
+        segments.push(if invalid_base_name(&base) {
+            Cow::Borrowed("*")
+        } else {
+            base
+        });
+    }
+
+    let mut joined = segments.join("/");
+    if !extension.is_empty() {
+        joined.push('.');
+        joined.push_str(extension);
+    }
+
+    joined
+}
+
+const MAX_SEGMENT_LENGTH: usize = 25;
+
+fn invalid_base_name(segment: &str) -> bool {
```

Comment:
I'd also consider alternating casing for a continuous 1 word segment as a sign of a random ID, disregarding length.